zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/ZakenApiPlugin.kt

```diff
@@ -331,6 +333,44 @@ class ZakenApiPlugin(
         )
     }
 
+    @PluginAction(
+        key = "start-hersteltermijn",
+        title = "Start hersteltermijn",
+        description = "Start the recovery period for a case",
+        activityTypes = [SERVICE_TASK_START, USER_TASK_CREATE]
+    )
+    @Transactional
+    open fun startHersteltermijn(
+        execution: DelegateExecution,
+        @PluginActionProperty maxDurationInDays: Int,
+    ) {
+        val documentId = UUID.fromString(execution.businessKey)
+        val zaakUrl = zaakUrlProvider.getZaakUrl(documentId)
+        val startDate = LocalDate.now()
+        val hersteltermijn = ZaakHersteltermijn(
+            zaakUrl = zaakUrl,
+            startDate = startDate,
+            maxDurationInDays = maxDurationInDays
+        )
+
+        val existingHerseltermijn = zaakHersteltermijnRepository.findByZaakUrlAndEndDateIsNull(zaakUrl)
+        check(existingHerseltermijn == null || existingHerseltermijn != hersteltermijn) { "Hersteltermijn already exists for zaak '$zaakUrl'. " }
+
+        if (existingHerseltermijn == null) {
+            val zaak = client.getZaak(authenticationPluginConfiguration, zaakUrl)
+            val uiterlijkeEinddatumAfdoening = zaak.uiterlijkeEinddatumAfdoening
+                ?: calculateUiterlijkeEinddatumAfdoening(zaak.zaaktype, zaak.startdatum)
+            require(uiterlijkeEinddatumAfdoening != null) { "No 'uiterlijkeEinddatumAfdoening' available for zaak '$zaakUrl' " }
+
+            client.patchZaak(
+                authenticationPluginConfiguration, url, PatchZaakRequest(
+                    uiterlijkeEinddatumAfdoening = uiterlijkeEinddatumAfdoening.plusDays(maxDurationInDays.toLong())
+                )
+            )
+            zaakHersteltermijnRepository.save(hersteltermijn)
```

Comment:
Please use a spring-managed service instead so you can do transactional stuff.

Reply:
I've changed it to use a platformTransactionManager